benchexec/tools/igml.py

```diff
@@ -0,0 +1,136 @@
+        base_dir = os.path.join(os.path.dirname(executable), os.path.pardir)
+        jar_file = os.path.join(base_dir, "igml.jar")
+        bin_dir = os.path.join(base_dir, "target")
+        src_dir = os.path.join(base_dir, "src")
+        return executable
+
+    def program_files(self, executable):
+        return self._program_files_from_executable(
+            executable, self.REQUIRED_PATHS, parent_dir=True
+        )
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def name(self):
+        return "IGML"
+
+    def _get_additional_options(self, existing_options, task, rlimits):
+        options = []
+        if rlimits.cputime and "-timelimit" not in existing_options:
+            options += ["-timelimit", str(rlimits.cputime)]
+
+        if task.property_file:
+            options += ["-p", task.property_file]
+
+        # try:
+        #     options += ["-f", task.single_input_file]
+        # except benchexec.tools.template.UnsupportedFeatureException:
+        #     raise benchexec.tools.template.UnsupportedFeatureException(
+        #         "Unsupported task with multiple input files for task '{}'".format(task)
+        # )
+
+        if isinstance(task.options, dict) and task.options.get("language") == "C":
+            data_model = task.options.get("data_model")
+            if data_model:
+                data_model_option = get_data_model_from_task(
+                    task, {ILP32: "-m 32", LP64: "-m 64"}
+                )
+                if data_model_option:
+                    if data_model_option not in existing_options:
+                        options += [data_model_option]
+                else:
+                    raise benchexec.tools.template.UnsupportedFeatureException(
+                        "Unsupported data_model '{}' defined for task '{}'".format(
+                            data_model, task
+                        )
+                    )
+        return options
+
+    def cmdline(self, executable, options, task, rlimits):
+        additional_options = self._get_additional_options(options, task, rlimits)
+        return [executable] + options + additional_options
+
+    def determine_result(self, run):
+        """
+        @return: status of IGML after executing a run
+        """
+
+        reason = ""
+        for line in run.output:
+            if "de.upb.cs.sms.igml.errors.LearningFailedException" in line:
+                reason = "(Learner failed)"
+            if "de.upb.cs.sms.igml.errors.DataGenerationFaildException" in line:
+                reason = "(Data generation failed)"
+            if "Error: Invalid configuration (Cannot handle programs with arrays!)" in line:
+                return result.RESULT_UNKNOWN + "(Arrays)"
+            if "org.sosy_lab.common.configuration.InvalidConfigurationException: " \
+               "The generation of the strongest post conditions failed!" in line:
```

Comment:
```suggestion
            if (
                "Error: Invalid configuration (Cannot handle programs with arrays!)"
                in line
            ):
                return result.RESULT_UNKNOWN + "(Arrays)"
            if (
                "org.sosy_lab.common.configuration.InvalidConfigurationException: "
                "The generation of the strongest post conditions failed!" in line
            ):
```
Formatting according to what our formatter black would do.

Reply:
Thanks for the hint :-)